### PR TITLE
Copy only relevant lines from http vhost to ssl vhost skeleton [needs revision]

### LINKED ISSUE #1996: the apache plugin adds an unconditional redirect inside the :443 vhost
The apache plugin adds an unconditional redirect inside the :443 vhost

I had to comment out these lines or else the webaddress just kept looping

RewriteEngine on
RewriteRule ^ https://%{SERVER_NAME}%{REQUEST_URI} [L,QSA,R=permanent]

### PR DESCRIPTION
**Pull request by sagi**
Closes #1996. See #1996 (comments) for reasoning.

### CODE REVIEW

**Comment by pde** [comment]
What if there's a rewrite from `/path/1` to `script.php?command=path&param=1`?  Wouldn't failing to copy the rewrite cause breakage in that case?

**Comment by pde** [comment]
Unless we can land a fixed-up version of this, it may be best to revert #1707 for 0.2.0 :(

File: letsencrypt-apache/letsencrypt_apache/configurator.py [hunk 1]
```diff
@@ -697,6 +697,47 @@ def _get_ssl_vhost_path(self, non_ssl_vh_fp):
...
+        Copying the above line to the ssl vhost would cause a redirection loop.
+
+        :param str line: a line extracted from the http vhost config file.
+
+        :returns: True - don't copy line from http vhost to SSL vhost.
+        :rtype: (bool)
+        """
+
+        rewrite_rule = "RewriteRule"
+        if line.lstrip()[:len(rewrite_rule)] != rewrite_rule:
```
Comment (pde) [inline letsencrypt-apache/letsencrypt_apache/configurator.py]: `if not line.lstrip.startswith(rewrite_rule):` is nicer

File: letsencrypt-apache/letsencrypt_apache/configurator.py [hunk 2]
```diff
@@ -697,6 +697,47 @@ def _get_ssl_vhost_path(self, non_ssl_vh_fp):
...
+        # line starts with RewriteRule.
+
+        # According to: http://httpd.apache.org/docs/2.4/rewrite/flags.html
+        # The syntax of a RewriteRule is:
+        # RewriteRule pattern target [Flag1,Flag2,Flag3]
+        # i.e. target is required, so it must exist.
+        target = line.split()[2].strip()
+
+        # target may be surrounded with double quotes
+        if len(target) > 0 and target[0] == target[-1] == "\"":
```
Comment (bmw) [inline letsencrypt-apache/letsencrypt_apache/configurator.py]: What other characters can be used here? At the very least, using single quotes seems to work.

File: letsencrypt-apache/letsencrypt_apache/configurator.py [hunk 3]
```diff
@@ -697,6 +697,47 @@ def _get_ssl_vhost_path(self, non_ssl_vh_fp):
...
+            target = target[1:-1]
+
+        https_prefix = "https://"
+        if len(target) < len(https_prefix):
```
Comment (pde) [inline letsencrypt-apache/letsencrypt_apache/configurator.py]:
```
if target.startswith(https_prefix):
    return True
return False
```

File: letsencrypt-apache/letsencrypt_apache/configurator.py [hunk 4]
```diff
@@ -697,6 +697,47 @@ def _get_ssl_vhost_path(self, non_ssl_vh_fp):
...
+            return False
+
+        if target[:len(https_prefix)] == https_prefix:
```
Comment (bmw) [inline letsencrypt-apache/letsencrypt_apache/configurator.py]: What if someone has a redirect to another HTTPS site? For example, what if someone wants to have both `http://example.com` and `https://example.com` redirect to `https://www.example.com`?

File: letsencrypt-apache/letsencrypt_apache/configurator.py [hunk 5]
```diff
@@ -714,9 +755,27 @@ def _copy_create_ssl_vhost_skeleton(self, avail_fp, ssl_fp):
             with open(avail_fp, "r") as orig_file:
                 with open(ssl_fp, "w") as new_file:
                     new_file.write("<IfModule mod_ssl.c>\n")
+                    sift = False
+
                     for line in orig_file:
-                        new_file.write(line)
+                        if self._sift_line(line):
+                            if not sift:
+                                new_file.write("# The following rewrite rules "
```
Comment (bmw) [inline letsencrypt-apache/letsencrypt_apache/configurator.py]: nit: From this wording, I'd expect all disabled rewrite rules to be found directly under this comment. This is only the case, however, if all disabled rewrite rules are adjacent in the file.

Comment (pde) [inline letsencrypt-apache/letsencrypt_apache/configurator.py]: "One or more of the following rewrite rules"?

**Comment by bmw** [comment]
@sagi, just so you're aware, we'd like this to land in `0.2.0` so I'm currently working on a PR based on this to make the small revisions pde and I suggested.

**Pull request merged**